tests/CMakeLists.txt

```diff
@@ -0,0 +1,106 @@
+cmake_minimum_required(VERSION 3.5)
+project(unittest)
+
+set(EXAMPLES "matmul;minimal" CACHE STRING "a list of examples to run as tests")
+set(SAMPLES hosvd matmul minimal seissol_eqspp stock tce)
+set_property(CACHE EXAMPLES PROPERTY STRINGS ${SAMPLES})
+
+set(VARIANT "Eigen" CACHE STRING "example-specific variant")
+set(VARIANTS Eigen LIBXSMM OpenBLAS)
+set_property(CACHE VARIANT PROPERTY STRINGS ${VARIANTS})
+
+set(PRECISION "double" CACHE STRING "floating point type: double/single")
+set(PRECISION_OPTIONS "double" "single")
+set_property(CACHE PRECISION PROPERTY STRINGS ${PRECISION_OPTIONS})
+
+set(ARCH "hsw" CACHE STRING "floating point type: double/single")
+set(ARCH_OPTIONS snb hsw skx thunderx2t99 nvidia)
+set_property(CACHE ARCH PROPERTY STRINGS ${ARCH_OPTIONS})
+
+# adjust cli arg. for the code gen.
+if (${PRECISION} STREQUAL "double")
+    set(ARCH_STR d${ARCH})
+else()
+    set(ARCH_STR s${ARCH})
+endif()
+
+# find appropriate compiler flags
+if (${ARCH} STREQUAL "snb")
+    set(CPU_FLAGS "-mavx")
+elseif (${ARCH} STREQUAL "hsw")
+    set(CPU_FLAGS "-march=core-avx2")
+elseif(${ARCH} STREQUAL "skx")
+    set(CPU_FLAGS "-xCORE-AVX512" "-fma")
+elseif(${ARCH} STREQUAL "thunderx2t99")
+    set(CPU_FLAGS "-thunderx2t99")
+else()
```

Comment:
That has to be wrong, right?

Reply:
right, it is a typo. I guess it should be
```
-mcpu=thunderx2t99
```